src/tools/perf/perftest.c

```diff
@@ -1376,14 +1386,24 @@ static ucs_status_t check_system(struct perftest_context *ctx)
     return UCS_OK;
 }
 
-static void clone_params(ucx_perf_params_t *dest, const ucx_perf_params_t *src)
+static ucs_status_t clone_params(ucx_perf_params_t *dest,
+                                 const ucx_perf_params_t *src)
 {
     size_t msg_size_list_size;
 
     *dest               = *src;
     msg_size_list_size  = dest->msg_size_cnt * sizeof(*dest->msg_size_list);
     dest->msg_size_list = malloc(msg_size_list_size);
-    memcpy(dest->msg_size_list, src->msg_size_list, msg_size_list_size);
+    if (dest->msg_size_list != NULL) {
```

Comment:
why not just if ==NULL return UCS_ERR_.. ?

Reply:
fixed
now it is `if ((est->msg_size_list == NULL) && (msg_size_list_size  != 0))`